tests/satosa/frontends/test_openid_connect.py

```diff
@@ -199,9 +291,64 @@ def test_provider_configuration_endpoint(self, context, frontend):
 
         provider_config_dict = provider_config.to_dict()
         scopes_supported = provider_config_dict.pop("scopes_supported")
+        assert "eduperson" not in scopes_supported
         assert all(scope in scopes_supported for scope in ["openid", "email"])
         assert provider_config_dict == expected_capabilities
 
+    def test_provider_configuration_endpoint_with_extra_scopes(
+        self, context, frontend_with_extra_scopes
+    ):
+        expected_capabilities = {
+            "response_types_supported": ["code", "id_token", "code id_token token"],
+            "jwks_uri": "{}/{}/jwks".format(BASE_URL, frontend_with_extra_scopes.name),
+            "authorization_endpoint": "{}/foo_backend/{}/authorization".format(
+                BASE_URL, frontend_with_extra_scopes.name
+            ),
+            "token_endpoint": "{}/{}/token".format(
+                BASE_URL, frontend_with_extra_scopes.name
+            ),
+            "userinfo_endpoint": "{}/{}/userinfo".format(
+                BASE_URL, frontend_with_extra_scopes.name
+            ),
+            "id_token_signing_alg_values_supported": ["RS256"],
+            "response_modes_supported": ["fragment", "query"],
+            "subject_types_supported": ["pairwise"],
+            "claim_types_supported": ["normal"],
+            "claims_parameter_supported": True,
+            "request_parameter_supported": False,
+            "request_uri_parameter_supported": False,
+            "claims_supported": [
+                "email",
+                "eduperson_scoped_affiliation",
+                "eduperson_principal_name",
+            ],
+            "grant_types_supported": ["authorization_code", "implicit"],
+            "issuer": BASE_URL,
+            "require_request_uri_registration": False,
+            "token_endpoint_auth_methods_supported": ["client_secret_basic"],
+            "version": "3.0",
+        }
+
+        http_response = frontend_with_extra_scopes.provider_config(context)
+        provider_config = ProviderConfigurationResponse().deserialize(
+            http_response.message, "json"
+        )
+
+        provider_config_dict = provider_config.to_dict()
+        scopes_supported = provider_config_dict.pop("scopes_supported")
+        assert all(
+            scope in scopes_supported for scope in ["openid", "email", "eduperson"]
+        )
+
+        # FIXME why is this needed?
+        expected_capabilities["claims_supported"] = set(
+            expected_capabilities["claims_supported"]
+        )
+        provider_config_dict["claims_supported"] = set(
+            provider_config_dict["claims_supported"]
+        )
```

Comment:
I am not sure why we need to do this, here. I added this as a workaround. Would you like to look into it more?

Reply:
Notice, this is not about the order of the dict-fields, but the order of the list elements. Lists are ordered, so this should not happen. This is ofcourse a serialized version of that object, and I'm not sure how the list is generated.

Reply:
Actually we do not need to check again provider_config against the expected_capabilities. We can just check the scopes_supported. The problem here is python 3.5 and the random order of the lists. I wonder though how it did not fail randomly also for the other lists in the dict.

Reply:
> The problem here is python 3.5 and the random order of the lists.

No, this is not correct - there is no "random order of the lists". What is was not ordered -and I still think one should not rely on that property- is the order of the keys of dictionaries. Lists are and always were ordered.